src/daft-dsl/src/functions/kv/kv_functions.rs

```diff
@@ -0,0 +1,331 @@
+    }
+
+    fn call(&self, args: FunctionArgs<Series>) -> DaftResult<Series> {
+        let inputs = args.into_inner();
+        if inputs.len() != 2 && inputs.len() != 3 {
+            return Err(DaftError::ValueError(
+                "kv_get_with_name expects 2 or 3 arguments".to_string(),
+            ));
+        }
+        let keys = &inputs[0];
+        let store_name_series = &inputs[1];
+        let columns_series_opt = if inputs.len() == 3 {
+            Some(&inputs[2])
+        } else {
+            None
+        };
+        if store_name_series.len() != 1 {
+            return Err(DaftError::ValueError(
+                "Store name series must contain exactly one element".to_string(),
+            ));
+        }
+        let name = store_name_series
+            .utf8()?
+            .get(0)
+            .ok_or_else(|| DaftError::ValueError("Missing store name".to_string()))?;
+
+        #[cfg(feature = "python")]
+        {
+            use daft_core::prelude::*;
+            // Parse columns if provided
+            let requested: Option<Vec<String>> = if let Some(cols_ser) = columns_series_opt {
+                let arr = cols_ser.binary()?;
+                let bytes = arr.get(0).ok_or_else(|| {
+                    DaftError::ValueError("columns must contain exactly one element".to_string())
+                })?;
+                let cols: Vec<String> = serde_json::from_slice(bytes).map_err(|e| {
+                    DaftError::ValueError(format!("Failed to deserialize columns: {}", e))
+                })?;
+                Some(cols)
+            } else {
+                None
+            };
+
+            let k_py = keys.cast(&DataType::Python)?;
+            let _k_arr = k_py.downcast::<PythonArray>()?;
+            let mut result_series_opt: Option<Series> = None;
+            Python::attach(|py| {
+                let daft_mod = py.import("daft.daft").unwrap();
+                let py_keys = pyo3::Py::new(
+                    py,
+                    daft_core::python::PySeries {
+                        series: keys.clone(),
+                    },
+                )
+                .unwrap();
+                let func = daft_mod.getattr("kv_get_direct_series").unwrap();
+                let py_res = match requested.clone() {
+                    Some(cols) => func.call1((name, py_keys, cols)).unwrap(),
+                    None => func.call1((name, py_keys, None::<Vec<String>>)).unwrap(),
+                };
+                let py_series: pyo3::PyRef<daft_core::python::PySeries> = py_res.extract().unwrap();
+                let mut s = py_series.series.clone();
+                s = s.rename(keys.name());
+                result_series_opt = Some(s);
+            });
+            let s = result_series_opt.ok_or_else(|| {
+                DaftError::ValueError("kv_get_direct_series returned None".to_string())
+            })?;
+            Ok(s)
+        }
+        #[cfg(not(feature = "python"))]
+        {
```

Comment:
i think we could just panic here. the `not(feature="python")` is really only ever enabled during testing.